code/modules/jobs/job_types/chaplain/chaplain_nullrod.dm

```diff
@@ -646,3 +690,172 @@
+		successful_sneak_attack = FALSE
+		sneak_attack_fail_message = TRUE
+
+	else if(IS_HERETIC_MONSTER(living_target) && prob(50)) // IT IS HARD TO SNEAK ATTACK SOMETHING WITH TOO MANY REDUNDANT EVERYTHINGS.
+		successful_sneak_attack = FALSE
+		sneak_attack_fail_message = TRUE
+
+	else if(HAS_TRAIT(living_target, TRAIT_STABLEHEART) && prob(50)) // THEIR ANATOMY IS FUCKING WEIRD.
+		successful_sneak_attack = FALSE
+		sneak_attack_fail_message = TRUE
+
+	else if(HAS_TRAIT(living_target, TRAIT_MIND_READER) && !user.can_block_magic(MAGIC_RESISTANCE_MIND, charge_cost = 0)) // FORESIGHT SAYS 'FUCK YOU' TO SNEAK ATTACKERS. BUT IF YOU HAVE A TIN FOIL HAT, YOU'RE SAFE!
+		successful_sneak_attack = FALSE
+		sneak_attack_fail_message = TRUE
+
+	else if(user.is_blind()) // YOU CAN'T STAB PRECISELY WHAT YOU CAN'T SEE.
+		successful_sneak_attack = FALSE
+		sneak_attack_fail_message = TRUE
+
+	/// And now we'll deal with sneak attack damage modifiers.
+
+	// If our target is also unconscious for some reason, we get even more damage. Coup de grace, motherfucker.
+	if(HAS_TRAIT(living_target, TRAIT_KNOCKEDOUT))
+		sneak_attack_dice += roll("1d6")
+		new /obj/effect/temp_visual/crit(get_turf(living_target))
+
+	// If the target is rebuked, we also add some additional damage. It is the closest thing to 'studying' your target, okay?
+	if(living_target.has_status_effect(/datum/status_effect/rebuked))
+		sneak_attack_dice += 2
+
+	// If we're morbid, and the target has been dissected, we get an extra d6.
+	// The chances of this occuring are quite low, as even having this weapon means you're locked out of becoming morbid as a chaplain, but when it does come up...
+	// Or the coroner stole this blade to go hunt the recently dead...
+	if(HAS_TRAIT(user, TRAIT_MORBID) && HAS_TRAIT(living_target, TRAIT_DISSECTED))
+		sneak_attack_dice += roll("1d6")
+
+	// Baton + this weapon might be a little too much fun so we're nerfing this combination outright.
+	if(HAS_TRAIT(living_target, TRAIT_IWASBATONED))
+		sneak_attack_dice *= 0.5
+
+	/// And so we return here if we are not entitled to a sneak attack.
+	if(!successful_sneak_attack)
+		if(sneak_attack_fail_message)
+			user.balloon_alert(living_target, "sneak attack avoided!")
+		return
+
+	// Affecting body part check.
+	var/obj/item/bodypart/affecting = living_target.get_bodypart(user.get_random_valid_zone(user.zone_selected))
+	// Target's armor value. Accounts for armor penetration even though we have no armour_penetration defined on the parent.
+	var/armor_block = living_target.run_armor_check(affecting, MELEE, armour_penetration = armour_penetration)
+
+	// We got a sneak attack!
+	living_target.apply_damage(round(sneak_attack_dice, 1), BRUTE, def_zone = affecting, blocked = armor_block, wound_bonus = bare_wound_bonus, sharpness = SHARP_EDGED)
```

Comment:
Rounding to 1 instead of `DAMAGE_PRECISION`? Doesn't matter to me just checking

Reply:
I didn't know the define existed :U